cmd/kaspawallet/utils/format_kas.go

```diff
@@ -14,3 +16,28 @@ func FormatKas(amount uint64) string {
 	}
 	return res
 }
+
+// Takes in a string representation of the Kas value to convert to Sompi
+func KasToSompi(amount string) (uint64, error) {
+	err := ValidateAmountFormat(amount)
+
+	if err != nil {
+		return 0, err
+	}
+
+	// after validation, amount can only be either an int OR
+	// a float with an int component and decimal places
+	parts := strings.Split(amount, ".")
+	amountStr := ""
+
+	if len(parts) == 2 {
+		amountStr = fmt.Sprintf("%s%-*s", parts[0], 8, parts[1]) // Padded with spaces at the end to fill for missing decimals: Sample "0.01234    "
```

Comment:
Use `math.Log10(SompiPerKaspa)` to determine amount of zeroes (and also add a test that checks `SompiPerKaspa` is a multiple of 10.